awsproviderlint/passes/AWSAT004/AWSAT004.go

```diff
@@ -0,0 +1,62 @@
+// Package AWSAT004 defines an Analyzer that checks for
+// hardcoded state hashes
+package AWSAT004
+
+import (
+	"go/ast"
+	"go/token"
+	"regexp"
+
+	"github.com/bflad/tfproviderlint/passes/commentignore"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+const Doc = `check for hardcoded state hashes
+
+The AWSAT004 analyzer reports hardcoded state hashes. Hardcoded state hashes
+are an unreliable way to specifically address state values since hashes may
+change over time, be inconsistent across partitions, and can be inadvertently
+changed by modifying configurations.
+`
+
+const analyzerName = "AWSAT004"
+
+var Analyzer = &analysis.Analyzer{
+	Name: analyzerName,
+	Doc:  Doc,
+	Requires: []*analysis.Analyzer{
+		commentignore.Analyzer,
+		inspect.Analyzer,
```

Comment:
To prevent false positives, this linter should be using the results of the [`testcheckresourceattrcallexpr` pass](https://pkg.go.dev/github.com/bflad/tfproviderlint@v0.14.0/passes/helper/resource/testcheckresourceattrcallexpr) and checking the second call argument, so we can be sure its limited to only those particular instances. 👍

Reply:
Fixed ✔️